provider/machinepool/machine_pool_resource.go

```diff
@@ -437,13 +517,27 @@ func (r *MachinePoolResource) Update(ctx context.Context, req resource.UpdateReq
 		return
 	}
 
+	diags = r.doUpdate(ctx, state, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save the state:
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *MachinePoolResource) doUpdate(ctx context.Context, state *MachinePoolState, plan *MachinePoolState) (diags diag.Diagnostics) {
```

Comment:
Can we be more explicit and return the value in each return state? Took me a while to understand what was returned...

Reply:
done